calculator.rb

```diff
@@ -0,0 +1,183 @@
+#Calculator program
+
+#define methods to carry out operations
+def addition(num1, num2)
+  return num1 + num2
+end
+
+def subtraction(num1, num2)
+  return num1 - num2
+end
+
+def multiplication(num1, num2)
+  return num1 * num2
+end
+
+def division(num1, num2)
+  return num1.to_f / num2.to_f
+end
+
+def exponent(num1, num2)
+  return num1 ** num2
+end
+
+def mod(num1, num2)
+  return num1 % num2
+end
+
+#checks to make sure the user entered a real number
+#old version that doesn't account for 0.00000, etc.
+=begin
+def number_check(user_input)
+  if user_input == "0" || user_input == "0.0"
+    return 0
+  elsif user_input.to_f == 0
+    return false
+  else
+    return user_input
+  end
+end
+
+
+#attempt at regular expression that didn't work the way I wanted
+def number_check(user_input)
+  if user_input =~ /[0-9]*[\.]?[0-9]+/
+    return user_input
+  else
+    return false
+  end
+end
+
+=end
+
+#String method to test if a string is numeric
+# https://mentalized.net/journal/2011/04/14/ruby-how-to-check-if-a-string-is-numeric/
+
+class String
+  def numeric?
+    Float(self) != nil rescue false
+  end
+end
+
+#checks that user inputted a real number
+def number_check(user_input)
+  if user_input.numeric?
+    return user_input
+  else
+    return false
+  end
+end
+
+
+#verifies user input and keeps prompting until a valid number is entered
+def verify(input)
+  until number_check(input)
+    puts "Please enter a number!"
+    input = gets.chomp
+  end
+  return input.to_f
+end
+
+#checks if the final answer should be an int or a Float
+def look_pretty(answer)
+  if answer == answer.to_i
+    return answer.to_i
+  else
+    return answer
+  end
+end
+
+#define methods to get user input and print results
+def add_it
```

Comment:
I agree that this _could_ be dried up, but for the purposes of the assignment, I think this is good! You are keeping the output and calculations separated from the output, and it remains easy to follow!